integration-tests/tests/src/tests/sync/open-behavior.ts

```diff
@@ -0,0 +1,658 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import { expect } from "chai";
+import Realm from "realm";
+import { generatePartition, randomVerifiableEmail } from "../../utils/generators";
+import { ObjectId } from "bson";
+import { importAppBefore } from "../../hooks";
+import { sleep } from "../../utils/sleep";
+
+const DogForSyncSchema = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "string?",
+  },
+};
+
+async function getRegisteredEmailPassCredentials(app: Realm.App) {
+  if (!app) {
+    throw new Error("No app supplied to 'getRegisteredEmailPassCredentials'");
+  }
+
+  const email = randomVerifiableEmail();
+  const password = "test1234567890";
+  // Create the user (see note in 'randomVerifiableEmail')
+  await app.emailPasswordAuth.registerUser({ email, password });
+
+  return Realm.Credentials.emailPassword(email, password);
+}
+
+describe("OpenBehaviour", function () {
+  importAppBefore("with-db");
+  it("static references are defined", () => {
+    expect(Realm.App.Sync.openLocalRealmBehavior).to.not.be.undefined;
+    expect(Realm.App.Sync.downloadBeforeOpenBehavior).to.not.be.undefined;
+  });
+  it("open synced realm with localRealmBehaviour", async function (this: AppContext) {
+    // NOTE: this test no longer runs with a logged out user.
+    // Reason: Error: User is no longer valid.
+
+    const partitionValue = generatePartition();
+    const user = await this.app.logIn(Realm.Credentials.anonymous());
+    const config = {
+      schema: [],
+      sync: {
+        user,
+        partitionValue,
+        newRealmFileBehavior: Realm.App.Sync.openLocalRealmBehavior,
+      },
+    };
+
+    expect(Realm.exists(config)).to.be.false;
+
+    const realm = await Realm.open(config);
+
+    expect(realm.path).to.not.be.undefined;
+
+    realm.close();
+    await user.logOut();
```

Comment:
We should probably create a `logOutAfterEach` action.  Seems possible one could forget to do this and then there would still be an open user session on the next test.

Reply:
think a separate hook like that would be good yeah, but for now `Realm.ClearTestState` both logs out all users and removes existing realms if I'm not mistaken, so should be good for now